Allow passing other arguments to interpolation functions

cancan101 opened the pull request:
Closes #10378

cancan101 committed:
ENH: GH10378 Pass kwargs to interpolation methods.

jreback commented:
can you add some tests?

sinhrks commented on pandas/core/common.py:
```diff
@@ -1734,7 +1735,7 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
                                     bounds_error=bounds_error)
```
Should pass ``**kwargs``, or  only support ``UnivariateSpline``? (Also #1749line)

cancan101 replied:
It should support more types. I only wrote one so far.

sinhrks commented:
Pls update docstring in ``generic.py``.

cancan101 commented:
Okay. I added tests and updated the docstring. In looking more closely I'm actually not sure what other interpolating methods should be passed the kwargs. Potentially `piecewise_polynomial_interpolate` but I don't really understand it (#10365). I think `interp1d` is getting all of the relevant variables.

jorisvandenbossche commented on pandas/core/generic.py:
```diff
@@ -2896,6 +2896,7 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
             Update the NDFrame in place if possible.
         downcast : optional, 'infer' or None, defaults to None
             Downcast dtypes if possible.
+        kwargs : a dictionary of keyword arguments passed into interpolating function.
```
This shouldn't be a dictionary I think? 
So just "Other keyword arguments are passed to the interpolating function"?

cancan101 replied:
Updated.

cancan101 commented:
Anything else to get this merged? And wow, travis-ci is SLOW at running these tests.

jreback commented:
see here: http://pandas.pydata.org/pandas-docs/stable/contributing.html

pls squash.

and travis seems quite chipper lately. We have a quite complicated setup, so takes a bit.

cancan101 commented:
Ok, I'll update the docs and then squash. I usually wait until code is :+1: before squashing to make reviewing easier.

jreback commented on pandas/tests/test_generic.py:
```diff
@@ -1365,6 +1365,23 @@ def test_spline(self):
 
+    def test_spline_extrapolate(self):
+        tm.skip_if_no_package('scipy', '0.15', 'setting ext on scipy.interpolate.UnivariateSpline')
+        s = Series([1, 2, 3, 4, np.nan, 6, np.nan])
```
any other methods take args?

cancan101 replied:
I talked about this here: https://github.com/pydata/pandas/pull/10383#issuecomment-114668678, I can add `kwargs` to `piecewise_polynomial_interpolate` but I am not sure how to test that method.

jreback replied:
ok, why don't you just run it, so will be a smoke test (which asserts that it didn't crash)

cancan101 replied:
I actually can't get that method to run without the exception I link to in that comment. There seems to be no current test coverage of this method.

jreback replied:
ahh I see. hmm, wonder if that has api changes in recent versions to break this somehow?

cancan101 replied:
LMK what you want to do about this.

jorisvandenbossche replied:
I think we can leave that for another PR (as there is an issue for this: #10365), if you don't directly know how to solve it

cancan101 replied:
I have no idea how to solve it :-)

cancan101 commented:
Squashed and docs.

cancan101 commented:
Anything else remaining to get this merged?

jreback commented:
thanks @cancan101
